### DISCUSSION TITLE
How can i change LoginForm to username & password?

### DISCUSSION THREAD

**Discussion by Michael-indochat** [refinedev/refine#5092] [Q&A] [answered]
I'm utilizing an authentication library called AuthPage, which by default employs email and password as the login fields. My project, however, only needs userName (no validation required) and password for login.

I've attempted to directly modify the LoginPage within node_modules, but this appears to be unfeasible. Are there alternative solutions to this? Is creating a custom LoginForm the way to go?

- Is there a way to meet my requirements without altering node_modules?
- Is crafting my own LoginForm a viable solution, or are there other options available?

**Comment by alicanerdurmaz** [comment, marked as answer]
Hello @Michael-indochat 

You can [swizzle](https://refine.dev/docs/packages/documentation/cli/#swizzle) `<AuthPage />` component and easily customize according to your needs.
